packages/plugin-commands-publishing/src/publish.ts

```diff
@@ -188,7 +188,11 @@ Do you want to continue?`,
         }
       }
 
-      const { status } = runNpm(opts.npmPath, ['publish', '--ignore-scripts', ...args])
+      const publishProjectDir = manifest.publishConfig?.directory
+      const publishProjectDirectory = typeof publishProjectDir === 'string' ? path.join(dir, publishProjectDir) : undefined
```

Comment:
maybe
```suggestion
      const publishProjectDir = path.join(dir, manifest.publishConfig?.directory ?? '.')
```

Reply:
publishConfig.anything is "unknown", is a type error

Reply:
well, you can add the type for that field. It is probably in `@pnpm/types`

Reply:
I implemented it like this:

```ts
const cwd = manifest.publishConfig?.directory ? path.join(dir, manifest.publishConfig.directory) : undefined

const { status } = runNpm(opts.npmPath, ['publish', '--ignore-scripts', ...args], {
  cwd,
})
```

Since I think it should be marginally better not needing to call "path.join"